sql/catalyst/src/main/scala/org/apache/spark/sql/types/HiveNullType.scala

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.types
+
+/**
+ * A hive null type for compatibility. These datatypes should only used for parsing,
+ * and should NOT be used anywhere else. Any instance of these data types should be
+ * replaced by a [[NullType]] before analysis.
+ */
+class HiveNullType private() extends DataType {
```

Comment:
I know the context, but can we name this `HiveVoidType` literally?

Reply:
Currently, the description is interpreted like "`hive null type` should be replaced by a NullType before analysis".

Reply:
`null` is a `value` and Hive exposes `void` as a type.

Reply:
> `null` is a `value` and Hive exposes `void` as a type.

You are right.